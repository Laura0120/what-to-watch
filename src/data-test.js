export const movie = {
  id: 1,
  poster: `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Macbeth.jpg`,
  preview: `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/macbeth.jpg`,
  backgroundImage: `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Macbeth.jpg`,
  backgroundColor: `#F1E9CE`,
  video: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
  previewVideo: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  title: `Macbeth`,
  rating: {
    ratingScore: 3.3,
    countVotesRating: 48798
  },
  director: `Justin Kurzel`,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  year: 2015,
  runtime: 113,
  genre: `Drama`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  isFavorite: false
};


export const noop = () => {};

export const comment = {
  comment: `A movie that will take you to another world full of emotions.`,
  date: `2020-11-03T13:38:44.769Z`,
  id: 1,
  rating: 8.6
};
