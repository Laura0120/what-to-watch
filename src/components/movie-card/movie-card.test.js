import React from 'react';
import renderer from 'react-test-renderer';

import MovieCard from './movie-card';
import {movie, noop} from '../../data-test';

it(`MovieCard is rendered correctly`, () => {
  const tree = renderer.create(
      <MovieCard
        movie={movie}
        onMovieClick={noop}
        onMouseOver={noop}
        onMouseOut={noop}
        renderPlayer={noop}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
