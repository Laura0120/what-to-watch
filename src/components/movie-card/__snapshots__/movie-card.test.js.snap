// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard is rendered correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  id={1}
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="#"
    >
      Macbeth
    </a>
  </h3>
</article>
`;
